apps/taskman/service.py

```diff
@@ -79,6 +83,19 @@ def __init__(self, token) -> None:
 
     def create_or_update_task(self, flaw: Flaw) -> Response:
         """Creates or updates a task using Flaw data"""
+        token_validation_url = f"{self.jira_conn._get_url('mypermissions')}?projectKey={JIRA_TASKMAN_PROJECT_KEY}"
+
+        # This request raises exception for unauthenticated users
+        permission_response = self.jira_conn._session.get(token_validation_url)
+
+        permissions = permission_response.json().get(
+            "permissions", {"CREATE_ISSUES": {"havePermission": False}}
+        )
+        if not permissions["CREATE_ISSUES"]["havePermission"]:
+            raise TaskWritePermissionsException(
+                f"Token is valid for {JIRA_TASKMAN_URL} but user doesn't have write permission in {JIRA_TASKMAN_PROJECT_KEY} project."
+            )
```

Comment:
These safeguards are great! I still a bit wonder how it could happen before that the `taskman` could fail eg. on invalid token but then we proceeded to the flaw creation. I assume that such failure would happen a few lines below during some consecutive Jira call and these safeguards does not change anything there. It should however solve the current issues so definitely good to proceed with it.